python/pyarrow/tests/test_fs.py

```diff
@@ -71,160 +310,169 @@ class Path:
                      pathlib.Path()]
     for path in invalid_paths:
         with pytest.raises(TypeError):
-            fs.create_dir(path)
+            fs.impl.create_dir(path)
 
 
-def test_get_target_stats(fs, tempdir, testpath):
-    aaa, aaa_ = testpath('a/aa/aaa'), tempdir / 'a' / 'aa' / 'aaa'
-    bb, bb_ = testpath('a/bb'), tempdir / 'a' / 'bb'
-    c, c_ = testpath('c.txt'), tempdir / 'c.txt'
+def test_get_target_stats(fs):
+    _aaa, aaa = fs.pathpair('a/aa/aaa/')
+    _bb, bb = fs.pathpair('a/bb')
+    _c, c = fs.pathpair('c.txt')
 
-    aaa_.mkdir(parents=True)
-    bb_.touch()
-    c_.write_bytes(b'test')
+    fs.mkdir(_aaa)
+    fs.touch(_bb)
+    fs.write_bytes(_c, b'test')
 
-    def mtime_almost_equal(fs_dt, pathlib_ts):
+    def mtime_almost_equal(a, b):
         # arrow's filesystem implementation truncates mtime to microsends
         # resolution whereas pathlib rounds
-        pathlib_dt = datetime.utcfromtimestamp(pathlib_ts)
-        difference = (fs_dt - pathlib_dt).total_seconds()
-        return abs(difference) <= 10**-6
+        diff = (a - b).total_seconds()
+        return abs(diff) <= 10**-6
 
-    aaa_stat, bb_stat, c_stat = fs.get_target_stats([aaa, bb, c])
+    aaa_stat, bb_stat, c_stat = fs.impl.get_target_stats([aaa, bb, c])
 
     assert aaa_stat.path == aaa
     assert 'aaa' in repr(aaa_stat)
-    assert aaa_stat.base_name == 'aaa'
     assert aaa_stat.extension == ''
-    assert aaa_stat.type == FileType.Directory
-    assert mtime_almost_equal(aaa_stat.mtime, aaa_.stat().st_mtime)
-    with pytest.raises(ValueError):
-        aaa_stat.size
+    assert mtime_almost_equal(aaa_stat.mtime, fs.mtime(_aaa))
+    # type is inconsistent base_name has a trailing slas for 'aaa' and 'aaa/'
```

Comment:
@pitrou if the directory's path ends with a slash then the file type is File, otherwise it is Directory, is this the expected outcome?

Reply:
In which situation exactly? For guidance, you can take a look at the C++ tests, they should tell you what is expected.